### ISSUE TITLE
Speak: audible feedback for insert/edit/remove link

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#1867]
When users complete an action, they can infer from the UI visual state the action result: success, failure, errors, etc. Any action that gives just a visual feedback should also use an audible feedback.

When inserting/editing/removing a link in the visual editor (and ideally in the text editor too), a meaningful message should be sent to the `wp.a11y.speak` live region. The current editor does that, but just when inserting a link:

![screen shot 2017-07-12 at 14 57 59](https://user-images.githubusercontent.com/1682452/28119401-08a28704-6715-11e7-8ee2-8bec2383bb2b.png)

See `speak( window.wpLinkL10n.linkInserted )` and `'linkInserted' => __( 'Link inserted.' )` in `wpLinkL10n`.
 
It would be nice to improve the current behavior an add messages also when removing or updating links. For inspiration, other web apps already do this. Google docs, for example, does it and also does announce many other actions:

![screen shot 2017-07-12 at 15 31 14](https://user-images.githubusercontent.com/1682452/28120319-fae62320-6717-11e7-9b49-a267814f9b06.png)

![screen shot 2017-07-12 at 15 29 55](https://user-images.githubusercontent.com/1682452/28120321-faf7463c-6717-11e7-9f27-118d1d0b1e7f.png)

![screen shot 2017-07-12 at 15 30 02](https://user-images.githubusercontent.com/1682452/28120320-faea7be6-6717-11e7-9b0d-95f2df42d051.png)

![screen shot 2017-07-12 at 15 25 21](https://user-images.githubusercontent.com/1682452/28120323-fb372db0-6717-11e7-8518-32c8151fdd5c.png)

**Comment by afercia** [comment]
Just checked again and there is now a `Link inserted.` message. It would be nice to add messages also for the other actions.

**Comment by aduth** [comment]
Remaining action items:

- `FormatToolbar#submitLink` needs to speak for `this.state.isEditing`

https://github.com/WordPress/gutenberg/blob/97818006d2ca7a239ea8bf0bc055b12d350b6515/blocks/rich-text/format-toolbar/index.js#L113-L119

- `FormatToolbar#dropLink` needs to speak

https://github.com/WordPress/gutenberg/blob/97818006d2ca7a239ea8bf0bc055b12d350b6515/blocks/rich-text/format-toolbar/index.js#L103-L106

**Comment by jeffpaul** [comment]
_This ticket was mentioned in [Slack](https://make.wordpress.org/chat/) in #core-editor by jeffpaul. [View the logs](https://wordpress.slack.com/archives/C02QB2JS7/p1519319829000441)._
